pyproject.toml

```diff
@@ -43,6 +45,7 @@
     "lefthook>=1.11.13",
 ]
   test = [
+    "numpy>=1.24",
```

Comment:
Any particular reason for 1.24? I can't really remember if we decided on following [SPEC 0](https://scientific-python.org/specs/spec-0000/). Cus if we did, then we could go skip ahead to `>=1.25` (and `>=1.26` in Q3 2025).

Reply:
I doubt that it will matter for us, but there were some typing that were fixed in numpy 1.25, and there was a backwards incompatible `numpy.exceptions` change I believe. 
See e.g. https://github.com/jorenham/optype/pull/320/files

Reply:
Bumped.